### DISCUSSION TITLE
Error in evaluating SUM formula with multiple grouped arguments

### DISCUSSION THREAD

**Discussion by carlot0820** [PHPOffice/PhpSpreadsheet#1950] [Q&A]
I am getting this error:

`Formula Error: Unexpected ,`

when evaluating 

`=SUM((A1,A2),(B1,B2))`

In Excel, the formula works fine. Any idea why this can't be handled?

**Comment by MarkBaker** [comment]
It's because the parser assumes that a bracket means one of two things:
- Either that it's wrapping the series of arguments for a function (e.g. `SUM(1,2,3)`)
- Or that it's part of an arithmetic expression (e.g. `(3+4)*5`)

In this case, it's neither. The brackets for the SUM() function are the outer brackets, the inner brackets have no context that the parser can recognise.
When it encounters the closing bracket, the parser backtracks looking for an operator to indicate that this is part of an arithmetic expression; but while `,` can be an operator when working with column ranges, it isn't in this context. Alternatively, if no operator is found, it expects a function token when it finds the corresponding opening bracket; but in this case, there is no function
The error message itself relates to the token immediately after the closing bracket that triggered it, which is the comma separator for the SUM function arguments.

For interest, MS Excel will allow you to use such an expression using cell references (e.g. `=SUM((A1,A2),(B1,B2))`); but will not allow you to bracket values together like that when they are absolute values (e.g. `=SUM((2,3),(4,5))`).

**Comment by oleibman** [comment]
This is related to issue #4656. It is not yet fixed by PR #4657, possibly for the same reason as https://github.com/PHPOffice/PhpSpreadsheet/issues/4656#issuecomment-3508912063.

**Comment by oleibman** [comment]
Recent changes to PR #4657 do appear to fix this problem.
